server/src/test/java/io/crate/execution/engine/collect/ShardCollectorProviderTest.java

```diff
@@ -47,20 +49,41 @@ public void assertNoShardEntriesLeftInShardCollectSource() throws Exception {
             try {
                 //noinspection unchecked
                 Map<ShardId, ShardCollectorProvider> shardMap = (Map<ShardId, ShardCollectorProvider>) shards.get(shardCollectSource);
-                assertThat(shardMap.size(), is(0));
+                assertThat(shardMap.size(), is(numberOfShards));
             } catch (IllegalAccessException e) {
                 throw new RuntimeException(e);
             }
         }
     }
 
+    @Test
+    public void testClosedIndicesHaveShardEntries() throws Exception {
+        var numberOfShards = randomIntBetween(1, 4);
+        execute("create table t(i int) clustered into ? shards with (number_of_replicas='0-all')", new Object[]{numberOfShards});
+        ensureGreen();
+        execute("alter table t close");
+        waitUntilShardOperationsFinished();
+        assertNumberOfShardEntriesInShardCollectSource(numberOfShards);
+    }
+
+    @Test
+    public void testClosedIndicesHaveShardEntriesAfterSelectFromSysShards() throws Exception {
```

Comment:
Could you please elaborate why this tests a different behaviour compared to `testClosedIndicesHaveShardEntries`?

Reply:
I didn't write the initial test and only migrated it and i can see your point. Maybe @jeeminso as the original author can you help here ?

Reply:
I was trying to create temporary `ShardCollectorProvider` for closed indices whenever `select * from sys.shards` is requested. So I wanted to make sure after a select statement, the `shards` map would not contain entries for closed indices, hence I wrote `testClosedIndicesHaveNoShardEntriesAfterSelectFromSysShards`. Since now `shards` map will track closed indices as well, this test case could be taken out.

Reply:
This makes sense. Thank you for your feedback.